src/components/modules/toolbar/inline.ts

```diff
@@ -362,20 +362,68 @@ export default class InlineToolbar extends Module {
     }
 
     const toolSettings = this.Editor.Tools.getToolSettings(currentBlock.name);
+    const inlineToolbarSettings = toolSettings[this.Editor.Tools.USER_SETTINGS.ENABLED_INLINE_TOOLS];
 
-    return toolSettings && toolSettings[this.Editor.Tools.USER_SETTINGS.ENABLED_INLINE_TOOLS];
+    /**
+     * Checks if either tool's own inlineToolbar property or config.inlineToolbar property is is an array.
+     * If both are nanything other than arrays, then no inlineToolbar will be displayed.
+     * As there is no way to know which tools or order of tools user wants.
+     */
+    const isInlineToolbarArray = Array.isArray(inlineToolbarSettings) || Array.isArray(this.config.inlineToolbar);
+
+    return toolSettings && isInlineToolbarArray;
+  }
+
+  /**
+   * Sort tools according to the given order in inlineToolbar property
+   *
+   * @param inlineToolbar - inlineToolbar order.
+   */
+  private sortTools(inlineToolbar: string[]): void {
+    /**
+     * Gets all the button nodes with class `ce-inline-tool`
+     */
+    const buttonNodes = this.nodes.buttons.querySelectorAll(`.${this.CSS.inlineToolButton}`);
+
+    /**
+     * creates an array of all the button nodes
+     */
+    const buttons = Array.from(buttonNodes);
+
+    /**
+     * sorts the buttons array according to the inlineToolbar property provided by user
+     */
+    inlineToolbar.forEach((toolName, i) => {
+      buttons.forEach((toolBtn, j) => {
+        const _toolBtn = toolBtn as HTMLElement;
+
+        if (j > 0 && toolName == _toolBtn.dataset.tool) {
+          const temp = buttons[i];
+
+          buttons[i] = buttons[j];
+
+          buttons[j] = temp;
+        }
+      });
+    });
+
+    /**
+     * appends all the button nodes in sorted order
+     */
+    buttons.forEach(async (btn) => {
+      this.nodes.buttons.appendChild(btn);
```

Comment:
I thinks it would be more logical to return sorted buttons order from this function and append them to container somewhere else

Reply:
check the latest commit